sof-ctl: Add support for handling IPC4 blobs

ujfalusi opened the pull request:
Hi,

Update the ABI header to match with the kernel's recent changes to define it's use with IPC4 blobs.
Add support for generating and handling IPC4 blobs to sof-ctl.

this PR should be able to replace #6893

lgirdwood commented on tools/ctl/ctl.c:
```diff
@@ -20,6 +20,13 @@
 
+/* Definitions for multiple IPCs */
+enum sof_ipc_type {
+	SOF_IPC,
```
Nitpick: but important.
SOF_IPC_TYPE_ prefix needed for all, IPC4 is generic and NOT Intel only.

lgirdwood replied:
We need same update in the kernel.

ujfalusi replied:
Sure, but we first need to agree on the change, hence I was asking.

lyakh commented on tools/ctl/ctl.c:
```diff
@@ -68,24 +80,27 @@ static void usage(char *name)
 	fprintf(stdout, " -c control name e.g.");
 	fprintf(stdout, " numid=22,name=\\\"EQIIR1.0 EQIIR\\\"\"\n");
 	fprintf(stdout, " -n control id e.g. 22\n");
+	fprintf(stdout, " -I IPC type (0 - IPC3, 1 - IPC4), defaults to 0\n");
```
I think xtensa-build-zephyr.py has a more intuitive convention for "-i" / "--ipc": "IPC4" to force selection

lgirdwood replied:
We should really say ```-I 3``` for IPC3 and ```-I 4``` for IPC4

ujfalusi commented:
Changes since v1:
- rename the ipc typ enums to: SOF_IPC_TYPE_3 and SOF_IPC_TYPE_4 (is this OK to adopt for the kernel?)
- Use -i to select IPC type and align the parameter with xtensa-build-zephyr.py: IPC4 selects IPC4 type, but also add option for IPC3 just for the sake of completeness.

ujfalusi commented:
Changes since v2:
- Removed a leftover debug print, sorry.

ujfalusi commented:
Changes since v3:
- Simplify the IPC type parameter: from IPC3|IPC4 to 3|4

lgirdwood commented:
@wszypelt this PR is for a tool not tested by CI. False positive ?
